sherpa/include/sherpa/astro/xspec_extension.hh

```diff
@@ -79,19 +79,10 @@ namespace sherpa { namespace astro { namespace xspec {
 typedef sherpa::Array< float, NPY_FLOAT > FloatArray;
 typedef float FloatArrayType;
 
-// Try and support the use of std::transform while still building
-// against C++-98 compilers.
-//
-#if __cplusplus > 199711L
-#define CONVERTARRAY(orig, out, npts)					\
+// Assume std::transform is available on any system we support
```

Comment:
The source fails to compile on my mac intel running clang++ 15.0.0 with std=c++98. 

    In file included from sherpa/astro/xspec/src/_xspec.cc:22:
    sherpa/include/sherpa/astro/xspec_extension.hh:631:11: error: expected expression
          CONVERTARRAY(this->ear, fear);
          ^
    sherpa/include/sherpa/astro/xspec_extension.hh:85:24: note: expanded from macro 'CONVERTARRAY'
                       [](const double val) -> FloatArrayType { return static_cast<FloatArrayType>(val); });

When compiling without your changes  my builds pick up the else case of the macro (ie. __cplusplus == 199711L) when defining CONVERTARRAY.  Replacing lines 82-85 with the following works for my build:

float d2f_convert (double dd) { return static_cast<float>(dd); }
#define CONVERTARRAY(orig, out)                                        \
        std::transform(std::begin(orig), std::end(orig), std::begin(out), d2f_convert);

Reply:
gah. But yay for testing!

Reply:
One question is whether we want to support building with `--std=c++98`? We don't really document the supported C++ or C versions, but I thought we were aiming for something like `--std=c++11`?

Reply:
Of course, the easiest solution is to just revert this particular change.

Reply:
On my machine, clang (15.0.0) defaults to using __cplusplus == 199711L (the c++98 standard) and ignores the CXXFLAGS env var. Direct specification of '-std=c++11' on the commandline appears successful from the clang commandline.   As such, I would not suggest not dropping support of c++98 standard until we have a compelling reason to do so (ie. we need to take advantage of features only supported in a newer standard). I'm OK with reverting the code which has the ifdef to not use lambda expressions (introduced in c++11) when compiling against c++98.

Reply:
So I've reverted this commit.